src/ansys/simai/core/data/global_coefficients_requests.py

```diff
@@ -0,0 +1,209 @@
+
+    def _compose_calculette_payload(
+        self,
+        gc_formula: str,
+        sample_metadata: dict[str, Any],
+        bc: list[str] = None,
+        surface_variables: list[str] = None,
+    ):
+        """Composes the payload for a calculette request."""
+
+        surface_vars_list = EXTRA_CALCULETTE_FIELDS
+        if surface_variables:
+            surface_vars_list += surface_variables
+
+        return {
+            "formula": gc_formula,
+            "bc_list": bc if bc else [],
+            "surface_field_list": [
+                fd
+                for fd in sample_metadata.get("surface").get("fields")
+                if fd.get("name") in surface_vars_list
+            ],
+            "volume_field_list": [],
+        }
+
+    @abstractmethod
+    def run(self):
+        """Abstract method to perform the customized required process."""
+
+
+GlobalCoefficientRequestType = TypeVar(
+    "GlobalCoefficientRequestType", bound=GlobalCoefficientRequest
+)
+
+
+class GlobalCoefficientRequestDirectory(Directory[GlobalCoefficientRequestType]):
+    """Provides methods for handling SSEs related to Global Coefficients."""
+
+    def get(self, item_id: str) -> GlobalCoefficientRequestType:
+        """Get a  by project ID.
+
+        Args:
+            item_id: ID of the Global Coefficient request.
+        """
+
+        return self._registry[item_id] if item_id not in self._registry else None
+
+    def _handle_sse_event(self, data):
+        gc_formula = data.get("target").get("formula")
+        action = data.get("target").get("action")
+        item_id: str = f"{data['target']['id']}-{action}-{gc_formula}"
+        if item_id not in self._registry:
+            logger.debug(
+                f"{self.__class__.__name__}: Ignoring event for unknown object id {item_id}"
+            )
+            return
+        item = self._registry[item_id]
+
+        item._handle_job_sse_event(data)
+
+
+class CheckGlobalCoefficient(GlobalCoefficientRequest):
+    """Verifies a Global Coefficient formula."""
+
+    def run(self):
+        """Performs a check-formula request."""
+        self._client._api.check_formula(self.project_id, self._calculette_payload)
+
+    def _handle_job_sse_event(self, data):
+        """Manage object's state according to SSE."""
+        logger.debug(f"Handling SSE job event for {self._classname} id {self.id}")
+
+        state: str = data.get("status")
+
+        target = data.get("target")
+
+        if state in PENDING_STATES:
+            logger.debug(f"{self._classname} id {self.id} set status pending")
+            self._set_is_pending()
+        elif state == "successful":
+            logger.debug(f"{self._classname} id {self.id} set status successful")
+            self._set_is_over()
+        elif state in ERROR_STATES:
+            self.fields[
+                "error"
+            ] = f"Verification of formula {target.get('formula')} failed with {data.get('reason')}"
```

Comment:
You're using the message 3 times (twice copied then you get it to log the error), you could put it in a variable instead.
Also wondering if we should put a default in the `data.get("reason")` but I don't have a good idea of default error message.

Reply:
I set a `error_message` variable and put `UNKNOWN ERROR` as the default for `reason`.